packages/turf-transform-translate/index.js

```diff
@@ -0,0 +1,171 @@
+var helpers = require('@turf/helpers');
+var invariant = require('@turf/invariant');
+var rhumbDestination = require('@turf/rhumb-destination');
+var point = helpers.point;
+var feature = helpers.feature;
+var polygon = helpers.polygon;
+var multiPoint = helpers.multiPoint;
+var lineString = helpers.lineString;
+var multiPolygon = helpers.multiPolygon;
+var multiLineString = helpers.multiLineString;
+var getCoords = invariant.getCoords;
+
+/**
+ * Moves any geojson Feature or Geometry of a specified distance on the provided direction angle.
+ *
+ * @name translate
+ * @param {GeoJSON} geojson object to be translated
+ * @param {number} distance length of the motion; negative values determine motion in opposite direction
+ * @param {number} direction of the motion; angle from North between -180 and 180 decimal degrees, positive clockwise
+ * @param {string} [units=kilometers] in which `distance` will be express; miles, kilometers, degrees, or radians
+ * @param {number} [zTranslation=0] length of the vertical motion, same unit of distance
+ * @param {boolean} [mutate=false] allows GeoJSON input to be mutated (significant performance increase if true)
+ * @returns {GeoJSON} the translated GeoJSON feature
+ * @example
+ * var poly = turf.polygon([[0,29],[3.5,29],[2.5,32],[0,29]]);
+ * var translatedPoly = turf.translate(poly, 100, 35);
+ *
+ * //addToMap
+ * translatedPoly.properties = {stroke: '#F00', 'stroke-width': 4};
+ * var addToMap = [poly, translatedPoly];
+ */
+module.exports = function (geojson, distance, direction, units, zTranslation, mutate) {
+    // Input validation
+    if (!geojson) throw new Error('geojson is required');
+    if (distance === undefined || distance === null || isNaN(distance)) throw new Error('distance is required');
+    if (zTranslation && typeof zTranslation !== 'number' && isNaN(zTranslation)) throw new Error('zTranslation is not a number');
+
+    // shortcut no-motion
+    if (distance === 0 && zTranslation === 0) return (geojson.type === 'Feature') ? geojson : feature(geojson);
+
+    if (direction === undefined || direction === null || isNaN(direction)) throw new Error('direction is required');
+
+    if (distance < 0) {
+        distance = -distance;
+        direction = -direction;
+    }
+    var properties = geojson.properties || {};
+
+    // Clone geojson to avoid side effects
+    if (mutate === false || mutate === undefined) geojson = JSON.parse(JSON.stringify(geojson));
+
+    var motionVector = getCoords(rhumbDestination([0, 0], distance, direction, units));
```

Comment:
@stebogit  Not sure if this has been discussed but this `motionVector` based translation approach is not doing what you may be expecting it to do. The rhumb line should be calculated from the given point not from the origin and then translated. Won't give the same answer. You should be able to apply the same logic from the scale approach and just apply the rhumb transformation in the `coordEach` loop.